xtc/src/ledger.rs

```diff
@@ -5,54 +5,63 @@ use ic_cdk::export::candid::{CandidType, Principal};
 use ic_cdk::*;
 use ic_cdk_macros::*;
 use serde::*;
-use std::collections::hash_map::Entry;
 use std::collections::HashMap;
 
-pub struct Ledger(HashMap<Principal, u64>);
+#[derive(Default)]
+pub struct Ledger {
+
+    // stores the cycle balance hold by the principal
+    balances: HashMap<Principal, u64>,
```

Comment:
if a struct is used heavily in a lot of other places in our code, then it is better to make it a named struct and not a tuple like struct